app/controllers/appointments_controller.rb

```diff
@@ -0,0 +1,70 @@
+class AppointmentsController < ApplicationController
+  before_action :authenticate_user
+  before_action :set_appointment, only: %i[show update destroy]
+
+  # GET /appointments
+  def index
+    # all current user's appointments for future dates
+    @appointments = current_user.appointments
+    @upcoming_appointments = @appointments.where('date > ?', Date.today)
```

Comment:
Let's keep our controllers thin by not doing this "where" querying here. ActiveRecords is the better place to do all these

Reply:
Thanks for the tip. I have refactored this to the model for Appointments